compiler/rustc_mir/src/interpret/validity.rs

```diff
@@ -26,23 +26,27 @@ use super::{
 
 macro_rules! throw_validation_failure {
     ($where:expr, { $( $what_fmt:expr ),+ } $( expected { $( $expected_fmt:expr ),+ } )?) => {{
-        let msg = rustc_middle::ty::print::with_no_trimmed_paths(|| {
+        let (path, msg) = rustc_middle::ty::print::with_no_trimmed_paths(|| {
```

Comment:
Why a single big let for both of them, and not two separate `let`s?

Reply:
it was just not to have two separate `rustc_middle::ty::print::with_no_trimmed_path` calls: I'm unsure whether the non-path expressions actually required being wrapped by it when they are serialized into strings (and was unsure of the costs), and so found it safer to not change this macro too much.

I can try changing this to have 2 `let`s but a single `with_no_trimmed_path()` call for the actual path, and see whether it changes the test output. Or have 2 `let`s and 2 calls. Let me know what you prefer and I'll do it.

Reply:
The `msg` part *should* not need that `with_no_trimmed_paths` thing, so maybe try two `let`s with just the `path` part having `with_no_trimmed_paths` and then the test suite will tell us if that works or not.

Reply:
alright, I'll try that tonight, thanks for the review !

Reply:
The ui test suite seems to agree that the `msg` doesn't need anything related to path trimming, so I've pushed a separate commit (to help with review) to address this point.